### DISCUSSION TITLE
Custom "special files"  for App Router

### DISCUSSION THREAD

**Discussion by Charismara** [vercel/next.js#74308] [Ideas]
### Goals

1. Providing an API to Devs that allow them to define custom "special files" (like `layout.tsx`) in the app router.

### Background

Example use-case: 

Creating a `breadcrumb.tsx` file on each route which gets joined and rendered in a specific slot in the root layout. 

Or

Creating a `sidebar.tsx` on some routes to automatically create an entry within a sidebar component within the root layout.

### Proposal

I have no idea how this could be implemented.

**Comment by Arinji2** [comment]
Not a fan of this, custom files with special meanings from codebase to codebase.. that will hurt
